### PR TITLE
types: add key option to sign and verify

### LINKED ISSUE #313: `key` option is not defined in the types
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.24.3

### Plugin version

7.2.2

### Node.js version

20.9.0

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

11 22H2

### Description

I was trying to use the `fastify.jwt` object to verify a JWT provided by the "Sign in with Apple" flow, and found an inconsistency with the types of `jwt.verify` and `jwt.sign`.

The README suggests that both these methods should take a `key` as a default option as outlined [here for sign](https://github.com/fastify/fastify-jwt/blob/master/README.md#sign) and [here for verify](https://github.com/fastify/fastify-jwt/blob/master/README.md#verify). However, I am getting a type error in my project.

Upon inspection of your [type definitions](https://github.com/fastify/fastify-jwt/blob/master/types/jwt.d.ts#L175-L181) I found that the options parameters [defined here](https://github.com/fastify/fastify-jwt/blob/master/types/jwt.d.ts#L121-L129) extend fast-jwt's [SignerOptions](https://github.com/nearform/fast-jwt/blob/master/src/index.d.ts#L96-L110) and [VerifierOptions](https://github.com/nearform/fast-jwt/blob/master/src/index.d.ts#L117-L135) respectively, however those definitions do not contain the `key` property.

When you use these options to call fast-jwt's `createVerifier` [here](https://github.com/fastify/fastify-jwt/blob/master/jwt.js#L330-L332) and `createSigner` [here](https://github.com/fastify/fastify-jwt/blob/master/jwt.js#L309-L311), they both extend the schema you are using and add the `key` property [here](https://github.com/nearform/fast-jwt/blob/master/src/index.d.ts#L142-L148).

### Steps to Reproduce

1. Open a TypeScript project with fastify
2. Register the plugin
```js
const fastify = require('fastify')()
fastify.register(require('@fastify/jwt'), {
  secret: 'supersecret'
})
```
2. Try to sign a token with a custom key
```js
const token = fastify.jwt.sign({ foo: 'bar' }, { key: 'newkey' })
```

It will work, but give type errors.

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by KirillTregubov**
Closes #313 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by gurgunday** [comment]
Can you add a test for the type?

**Comment by KirillTregubov** [comment]
I have added tests to the `jwt.test-d.ts` file.

Interestingly, while looking around in the codebase, I could not find a justification for the `onlyCookie` definition on the `VerifyOptions` type. It is not documented in the README, and onlyCookie does not seem to be used in the `verify` function inside `jwt.js`. Please let me know if this is safe to remove.

File: types/jwt.d.ts [hunk 1]
```diff
@@ -121,11 +121,13 @@ declare namespace fastifyJwt {
   export interface SignOptions extends Omit<SignerOptions, "expiresIn" | "notBefore"> {
     expiresIn: number | string;
     notBefore: number | string;
+    key: string | Buffer
```
Comment (Eomm) [inline types/jwt.d.ts:124]: it seems optional to me - doesn't it?

```suggestion
    key?: string | Buffer

```

Comment (KirillTregubov) [reply types/jwt.d.ts:124]: Yes! Thank you for catching this, my mistake 😥

**Review by Eomm — APPROVED** [review]

File: types/jwt.d.ts (hunk 1 above)
Comment (Eomm) [reply types/jwt.d.ts]: No problem 😄

**Review by gurgunday — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**